### PR TITLE
Change naming of batch and epoch in callbacks

### PR DESCRIPTION
**Pull request by davebulaval**
I think the naming batch is misleading since it's a more a `batch_number` more than the usual `batch` concept.

### CODE REVIEW

**Comment by davebulaval** [comment]
After reading all the time batch and epoch was use in the doc, I think it would be more clear to use `batch_number` and `epoch_number` instead of verbally spoken name `step`.

**Comment by davebulaval** [comment]
Also, as a discussion, I don't clearly see the reason of having a the `batch` argument if the logs contain the actual `batch_number`.

File: poutyne/framework/callbacks/logger.py [hunk 1]
```diff
@@ -123,13 +124,16 @@ def __init__(self, writer):
         super().__init__(batch_granularity=False)
         self.writer = writer
 
-    def _on_batch_end_write(self, batch, logs):
+    def on_batch_end(self, batch_number, logs):
+        self._on_batch_end_write(batch_number, logs)
```
Comment (TortillasAlfred) [inline poutyne/framework/callbacks/logger.py]: Why add a on_batch_end() implementation here ?

**Comment by TortillasAlfred** [comment]
> After reading all the time batch and epoch was use in the doc, I think it would be more clear to use `batch_number` and `epoch_number` instead of verbally spoken name `step`.

Not sure what you mean here, any example of what you would change ?

**Comment by TortillasAlfred** [comment]
> Also, as a discussion, I don't clearly see the reason of having a the `batch` argument if the logs contain the actual `batch_number`.
> @freud14 @TortillasAlfred

I guess it is to ease reimplementation of ``on_batch_end()`` and other such methods. In the end, I do not see this as much of a big clean coding issue and the clarity gain, albeit small, is surely more impactful than the clean code violation.

File: poutyne/framework/callbacks/logger.py (hunk 1 above)
Comment (davebulaval) [reply poutyne/framework/callbacks/logger.py]: This is a mistake. I was working on a implementation of a gradient tracker.

**Comment by davebulaval** [comment]
> > After reading all the time batch and epoch was use in the doc, I think it would be more clear to use `batch_number` and `epoch_number` instead of as verbally spoken name `step`.
> 
> Not sure what you mean here, any example of what you would change ?

@freud14 and I spoke about the naming and we agreed on step, but as I was reading the doc, we were always referring as `batch: The batch number`. So I now disagreed with the name `step` for batch .

File: poutyne/framework/callbacks/logger.py (hunk 1 above)
Comment (davebulaval) [reply poutyne/framework/callbacks/logger.py]: af3815a

**Comment by freud14** [comment]
I'm not sure I see the issue with the current code and thus why this PR is necessary.

**Comment by davebulaval** [comment]
Similar as PR #53 where naming seem confusing in the callback.

**Comment by davebulaval** [comment]
It's just that the naming is misleading. Also added the close of the summary writer as similar of the  `CSVLogger`.

**Pull request merged**